soroban-sdk/src/env.rs

```diff
@@ -210,16 +209,25 @@ impl Env {
         Crypto::new(self)
     }
 
+    /// Get the Address object corresponding to the current executing contract.
+    pub fn current_contract_address(&self) -> Address {
+        let address = internal::Env::get_current_contract_address(self).unwrap_infallible();
+        unsafe { Address::unchecked_new(self.clone(), address) }
+    }
+
     /// Get the 32-byte hash identifier of the current executing contract.
-    pub fn current_contract(&self) -> BytesN<32> {
-        let id = internal::Env::get_current_contract(self).unwrap_infallible();
+    ///
+    /// Prefer `current_contract_address` for the most cases, unless dealing
+    /// with contract-specific functions (like the call stacks).
+    pub fn current_contract_id(&self) -> BytesN<32> {
+        let id = internal::Env::get_current_contract_id(self).unwrap_infallible();
```

Comment:
It's unfortunate that we have to have both of these functions. As a user I'm not sure which one I should use. I'm likely to choose the ID in situations where I should use the Adddress. I'm not sure what the solution to that is though.